internal/gateway/gateway_instance.go

```diff
@@ -603,31 +588,89 @@ func confirmGatewayPortExposed(
 		return fmt.Errorf("failed to update gateway controller workload instance: %w", err)
 	}
 
+	// log the ports that are exposed
+	gatewayPorts, err := client.GetGatewayPortsAsString(r.APIClient, r.APIServer, *gatewayInstance.GatewayDefinitionID)
+	if err != nil {
+		return fmt.Errorf("failed to get gateway ports as string: %w", err)
+	}
 	log.V(1).Info(
 		"updated gateway controller instance to expose requested port",
-		"port", fmt.Sprintf("%d", *gatewayDefinition.TCPPort),
+		"ports", fmt.Sprintf("%s", gatewayPorts),
 	)
 
 	return nil
 
 }
 
-// configureVirtualService configures a VirtualService custom resource
+// ensureGlooEdgePortExists ensures a gloo edge port exists
+// for a given protocol, port and tls state.
+func ensureGlooEdgePortExists(protocol string, port int, tlsEnabled bool, ports []interface{}, log *logr.Logger) ([]interface{}, error) {
+
+	// check existing gateways for requested ports
+	for _, portSpec := range ports {
+		spec := portSpec.(map[string]interface{})
+		var portCurrent int64
+		portCurrent, portNumberFound, err := unstructured.NestedInt64(spec, "port")
+		if err != nil {
+			portCurrentFloat, _, errFloat := unstructured.NestedFloat64(spec, "port")
+			portCurrent = int64(portCurrentFloat)
```

Comment:
The port number is either stored as an `int64` or a `flaot64` depending on how the `GatewayDefinition` is created. E2E tests result in one outcome, while creating via a local config file results in another.

Reply:
Add util.NestedIntOrFloat()